File: rt/imageio/mrc.py
#!/usr/bin/python

from EMAN2 import *
import sys

infile = sys.argv[1]

e = EMData()
e.read_image(infile)
e.set_attr_dict("MRC.label3", EMObject("liwei peng"))
e.write_image("a.mrc", 0, MRC)

'''
e2 = EMData()
e2.read_image("a.mrc")
d = e2.get_attr_dict()
nlabels = d["MRC.nlabels"]
print "nlabels = ", nlabels

for i in range(int(nlabels)):
    label = "MRC.nlabel" + str(i)
    print d[label]
    
'''

File: rt/imageio/mrc.cpp
#include "emdata.h"

using namespace EMAN;

int main(int argc, char *argv[])
{
    if (argc != 2) {
	printf("usage: %s imagefile\n", argv[0]);
	exit(1);
    }
    
    EMData e;
    e.read_image(argv[1]);
    e.set_attr_dict("MRC.label1", "Liwei Peng");
    e.write_image("a.mrc", 0, EMUtil::IMAGE_MRC);
    return 0;
}
